src/raftstore/store/fsm/peer.rs

```diff
@@ -1733,14 +1743,49 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         Some(ready_to_merge)
     }
 
+    fn on_ready_catch_up_logs(
+        &mut self,
+        mut merge: CommitMergeRequest,
+        logs_up_to_date: Arc<AtomicU64>,
+    ) {
+        let region_id = self.fsm.region_id();
+        assert_eq!(region_id, merge.get_source().get_id());
+        self.fsm.peer.catch_up_logs = Some(logs_up_to_date);
+
+        let entries = merge.take_entries();
+        let first = entries.first().unwrap();
+        let last = entries.last().unwrap();
+
+        let log_idx = first.get_index() - 1;
+        let log_term = self.fsm.peer.get_index_term(log_idx);
+
+        // warp these entries as MsgAppend
+        let mut msg = eraftpb::Message::new();
+        msg.set_msg_type(MessageType::MsgAppend);
+        msg.set_from(self.fsm.peer.leader_id());
```

Comment:
PTAL @BusyJay

Reply:
leader id may be 0, is it OK?

Reply:
How the raft behave when stepping a message with zero leader id

Reply:
It may break everything.